packages/bun-usockets/src/bsd.c

```diff
@@ -446,24 +446,49 @@ LIBUS_SOCKET_DESCRIPTOR bsd_create_listen_socket(const char *host, int port, int
 
     LIBUS_SOCKET_DESCRIPTOR listenFd = LIBUS_SOCKET_ERROR;
     struct addrinfo *listenAddr;
-    for (struct addrinfo *a = result; a && listenFd == LIBUS_SOCKET_ERROR; a = a->ai_next) {
+    for (struct addrinfo *a = result; a != NULL; a = a->ai_next) {
         if (a->ai_family == AF_INET6) {
             listenFd = bsd_create_socket(a->ai_family, a->ai_socktype, a->ai_protocol);
             listenAddr = a;
+
+            if (_bsd_create_listen_socket(listenFd, listenAddr, port, options) != LIBUS_SOCKET_ERROR) {
```

Comment:
should we also have a check for `listenFd != LIBUS_SOCKET_ERROR`?

Reply:
before calling the function?

Reply:
Yes, fixed.